docs/content/topics/project-developers/secrets.md

```diff
@@ -9,239 +9,157 @@ aliases:
+of the secrets defined on our project.
 
-```javascript
-// THIS IS NOT SAFE! IT LEAKS YOUR SECRET INTO THE LOGS.
-const {events, Job, Group} = require("brigadier")
-
-events.on("push", function(e, project) {
-  console.log("My DB password is " + project.secrets.dbPassword)
-})
-```
-
-Secrets can be selectively passed to jobs by setting environment variables.
+This first example demonstrates direct access of a secret defined on a project
+in the event handler code outside of any Job:
 
 ```javascript
 // THIS IS NOT SAFE! IT LEAKS YOUR SECRET INTO THE LOGS.
-const {events, Job, Group} = require("brigadier")
-
-events.on("push", function(e, project) {
-  var j1 = new Job("secrets", "alpine:3.4")
+const { events } = require("@brigadecore/brigadier");
 
-  // Send the password to an environment variable.
-  j1.env = {
-    "DB_PASSWORD": project.secrets.dbPassword
-  }
-
-  // Print the env var to the log.
-  j1.tasks = [
-    "echo $DB_PASSWORD"
-  ]
+events.on("brigade.sh/cli", "exec", async event => {
+  console.log("Project secret foo = " + event.project.secrets.foo);
+});
 
-  j1.run()
-}
+events.process();
 ```
 
-In this case, we retrieve the secret from the project, and we pass it into the new
-Job as an environment variable. When the job executes, it can access the `dbPassword`
-as `$DB_PASSWORD`.
-
-Note that behind the scenes, Brigade is storing the environment variables in another
-Job-specific secret.
-
-## FAQ
-
-**Why don't all jobs get access to all of the secrets? Why do I have to pass them
-to the `Job.env`?**
-
-Brigade is designed to use off-the-shelf Docker images. In the examples above, we used the
-`alpine:3.4` image straight from DockerHub. We wouldn't want to just automatically pass
-all of our information straight into that container. For starters, doing so might
-inadvertently override an existing environment variable of the same name. More
-importantly, the data might get misused or unintentionally exposed by the container.
-
-So we err on the side of safety.
-
-**Can I encrypt my secrets?**
-
-We use Kubernetes Secrets for holding sensitive data. As encrypted Secrets are
-adopted into Kubernetes, we plan to support them. However, the present stable
-version of Kubernetes Secrets only Base64-encodes data.
-
-Our present recommendation is for Brigade developers to fetch the secret directly from a
-trusted key store such as Vault.  See the [example](#fetching-secrets-from-a-remote-store) below.
-
-Alternatively, you could use `secretKeyRef` to reference existing secrets already in your
-Kubernetes cluster.  See the [example](#using-kubernetes-secrets) below.
-
-**I don't want to use Helm to manage my project/secrets. Can I do it manually?**
+For passing project secrets into a Job, the best practice is to do so via the
+Job's environment. Brigade inherently treats all Job environment variables as
+sensitive by default, so any instance where their values would otherwise be
+logged will be redacted.
```

Comment:
This statement was confusing. We don't _actively_ redact secrets whatever they might appear in the logs. We do store every job's environment in a k8s secret, but that is the extent of the measures we take. It's the user's responsibility to be careful about how they use their secrets.